app/instrumentation-tests/src/org/commcare/utils/InstrumentationUtility.kt

```diff
@@ -323,6 +324,13 @@ object InstrumentationUtility {
                 .perform(click())
     }
 
+    @JvmStatic
+    fun assertCurrentActivity(check: (Activity) -> Boolean) {
```

Comment:
you can just paas the activity::class here and localize the instance check in this function as well so that callers to this function just need to paas the class name for the activity

Reply:
I don't really get it. Do you mean to change this to something like
```
inline fun <reified T: Activity> assertCurrentActivity() {
        val application = InstrumentationRegistry.getInstrumentation().targetContext.applicationContext
                as CommCareInstrumentationTestApplication
        var activity = application.currentActivity
        assert(activity is T, "Current Activity is ${activity.localClassName}")
    }
```
and have tests call it as `InstrumentationUtility.assertCurrentActivity<LoginActivity>()`

Reply:
You can pass the class name as an argument to the function and check whether the current Activity is instance of that class - `fun assertCurrentActivity(Class<T> c)`